File: protocol/src/test/lua/zfoolua/packet/EmptyObject.lua

local EmptyObject = {}

function EmptyObject:new()
    local obj = {
        
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function EmptyObject:protocolId()
    return 0
end

function EmptyObject:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
end

function EmptyObject:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = EmptyObject:new()
    
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return EmptyObject

File: protocol/src/test/javascript/zfoojs/packet/EmptyObject.js

const EmptyObject = function() {
    
};

EmptyObject.PROTOCOL_ID = 0;

EmptyObject.prototype.protocolId = function() {
    return EmptyObject.PROTOCOL_ID;
};

EmptyObject.write = function(buffer, packet) {
    if (packet === null) {
        buffer.writeInt(0);
        return;
    }
    buffer.writeInt(-1);
};

EmptyObject.read = function(buffer) {
    const length = buffer.readInt();
    if (length === 0) {
        return null;
    }
    const beforeReadIndex = buffer.getReadOffset();
    const packet = new EmptyObject();
    
    if (length > 0) {
        buffer.setReadOffset(beforeReadIndex + length);
    }
    return packet;
};

export default EmptyObject;